plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -636,12 +636,10 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
                                 columns::get,
                                 columnId -> {
                                     IcebergColumnHandle column = columns.get(columnId);
-                                    Object prestoValue = deserializePartitionValue(
-                                            column.getType(),
-                                            partitionColumnValueStrings.get(columnId).orElse(null),
-                                            column.getName());
-
-                                    return NullableValue.of(column.getType(), prestoValue);
```

Comment:
Did you consider calling the `NullableValue` constructor instead?

Reply:
Good question. I don't have a preference. I don't quite understand why it has both constructor and static methods.
Which way do you think is better?

Reply:
they have different semantics -- `NullableValue.of` doesn't allow null value

(i with the difference was more obvious from the NullableValue class's api)